.github/actions/test-asdf-caf-scripts/action.yaml

```diff
@@ -0,0 +1,50 @@
+name: 'Test ASDF CAF Scripts'
+description: 'Install ASDF and required plugins and test functionality of CAF scripts'
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Install ASDF
+      if: ${{ inputs.install_asdf == 'true' }}
+      shell: bash
+      run: |
+        git clone https://github.com/asdf-vm/asdf.git ${{ inputs.asdf_dir }} --branch ${{ inputs.asdf_version }}
+
+    - name: Install Plugins
+      shell: bash
+      run: |
+        . "${{ inputs.ASDF_DIR }}/asdf.sh"
+        asdf plugin add caf-scripts "https://github.com/weareadaptive/asdf-caf-scripts"
+        echo -e "\ncaf-scripts ref:${{ github.sha }}" >> .tool-versions
```

Comment:
I think you can skip writing `.tool-versions` by setting an env var `ASDF_CAF_SCRIPTS_VERSION=${{ github.sha }}`. 

Then asdf install should pick it up. Example: https://github.com/AdaptiveConsulting/platform-exchange/blob/develop/.envrc#L28